src/scalar_mul.rs

```diff
@@ -67,31 +63,53 @@ pub trait VartimeMultiscalarMul {
     }
 }
 
-impl NonAdjacentForm for jubjub::Scalar {
-    /// Compute a width-\\(w\\) "Non-Adjacent Form" of this scalar.
+/// Produces the non-adjacent form (NAF) of a 32-byte scalar.
+pub trait NonAdjacentForm {
+    /// Returns the scalar represented as a little-endian byte array.
+    fn inner_to_bytes(&self) -> [u8; 32];
```

Comment:
If this is changed to `Vec<u8>` then I think the entire thing should work for scalars larger than 256 bits, which is something we'll need on FROST. But we don't need it here, so feel free to ignore for now

Reply:
It depends on the scalar size. There's this conversion, for example, https://github.com/zcashfoundation/reddsa/blob/f50ff9dd8ad4c79a561f68c28e6bac23790bbb32/src/scalar_mul.rs#L101-L102.

The algorithm should be easy to adjust, though. Plus, we have solid tests now.

Reply:
Reverted in [f52607e](https://github.com/ZcashFoundation/reddsa/pull/63/commits/f52607e3fc1675c2262f88f8612f7a90745c8b77).